tests/page-objects/forms/generic-form.po.js

```diff
@@ -45,6 +45,21 @@ module.exports = {
     }
   },
 
+  nextPageNative: async multiple => {
+    const nextButton = element(by.css('button.btn.btn-primary.next-page'));
+    if (multiple) {
+      for (let i = 0; i < multiple; i++) {
+        await helper.waitForAngularComplete();
+        await helper.waitElementToBeClickable(nextButton);
+        await nextButton.click();
+      }
+    } else {
+      await helper.waitUntilReady(nextButton);
+      await helper.waitElementToBeClickable(nextButton);
+      await nextButton.click();
+    }
+  },
```

Comment:
The conditional logic and structure here isn't immediately clear what differs. Reading in more detail, I see that it is mostly only about looping or not. We can even reword that as "how many times to loop" since we either need to loop for 1 time or more times. There are still some differences in the branches, but I think they're just waiting on things that should work fine either way and not incur any actual delays under normal operation. That means we can rewrite it to this:
```
    const iterations = multiple || 1;
    for (let i = 0; i < iterations; i++) {
        await helper.waitForAngularComplete();
        await helper.waitUntilReady(nextButton);
        await helper.waitElementToBeClickable(nextButton);
        await nextButton.click();
    }
  },
```

I'd also suggest changing `multiple` since one way of reading it might be as a boolean (and it is indeed evaluated for thruthiness). Fortunately, JavaScript can fix that up for us with a [default param value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters). With that, we don't even need that `multiple || 1` conditional logic anymore (wonderful for my allergy to conditional statements)! The param just becomes `iterations = 1` and we're good to go (still depending on my previous assumption on the waiting business there).

Reply:
I've reworked this logic. Thanks for catching that.